core/src/main/java/apoc/meta/Meta.java

```diff
@@ -642,27 +642,24 @@ private Tables4LabelsProfile collectTables4LabelsProfile (MetaConfig config) {
                 long labelCount = countStore.get(labelName);
                 long sample = getSampleForLabelCount(labelCount, config.getSample());
 
-                //System.out.println("Sampling " + sample + " for " + labelName);
-
                 try (ResourceIterator<Node> nodes = tx.findNodes(label)) {
                     int count = 1;
                     while (nodes.hasNext()) {
                         Node node = nodes.next();
-                        if(count++ % sample == 0) {
-                            boolean skipNode = false;
-                            for (RelationshipType rel : node.getRelationshipTypes()) {
-                                String relName = rel.name();
-                                if (excludeRels.contains(relName)) {
-                                    // Skip if explicitly excluded
-                                    skipNode = true;
-                                } else if (!includeRels.isEmpty() && !includeRels.contains(relName)) {
-                                    // Skip if included set is specified and this is not in it.
-                                    skipNode = true;
-                                }
-                            }
-                            if (skipNode != true) {
-                                profile.observe(node, config);
-                            }
+                        if (count++ % sample == 0) {
+                            final Set<Boolean> skips = StreamSupport
+                                    // we analyze the node for each its relationship type
+                                    .stream(node.getRelationshipTypes().spliterator(), false)
+                                    .map(rel -> excludeRels.contains(rel.name()) // we skip a node when the user said that must be excluded
+                                            // or when the user provided and inclusion list, but it's not in the provided list
+                                            || (!includeRels.isEmpty() && !includeRels.contains(rel.name())))
+                                    .collect(Collectors.toSet());
+                            // if the Set has just one element and is true we skip the node
+                            // if there are two elements [true, false] we don't skip it as give it means that
+                            // it have a relationship that satisfies the condition provided
+                            // by the configuration
+                            if (skips.size() == 1 && skips.iterator().next()) continue;
```

Comment:
I think this is not quite right. Can't it happen that we have multiple elements and they are all true, like `[true, true]`? In that case we wouldn't skip now, but I think perhaps we should? Maybe we could add some test around such a case?

Reply:
No `skips` collection is a `Set` so it contains only unique values like `[true, false]` or `[false, true]`, the collection will store only one for each boolean type.

Reply:
Ah, for some reason I thought it was a list and not a set, despite the comment. You are right!